src/dataExplorer/context/newDataExplorer.tsx

```diff
@@ -8,14 +15,17 @@ import {TagsContext} from 'src/dataExplorer/context/tags'
 // Types
 import {Bucket} from 'src/types'
 
-export const IMPORT_REGEXP = 'import "regexp"\n'
-export const IMPORT_INFLUX_SCHEMA = 'import "influxdata/influxdb/schema"'
-export const SAMPLE_DATA_SET = (bucketID: string) =>
-  `import "influxdata/influxdb/sample"\nsample.data(set: "${bucketID}")`
-export const FROM_BUCKET = (bucketName: string) =>
-  `from(bucket: "${bucketName}")`
+const DEBOUNCE_TIMEOUT = 500
+let timer
+type NOOP = () => void
+const debouncer = (action: NOOP): void => {
```

Comment:
not for this PR, just for your reference... i dont think this is _technically_ a debouncer. we have a bunch of message throttle controls we just refer to as a debouncer. there's a bunch of specific names for them that better describe their behavior if you wanna go look them up. this one just waits for a lull in the user activity and forces the `action` function to hold a sense of the source of truth (proxy for the input event always returning the WHOLE value). this simplistic model helps us get around weird cases where you deleted something, but the last update event in the debouncer was to add something, and the user sees ghosts typing in their input boxes! here's an even more complicated one that combines messages instead of dropping them like this one does:
https://github.com/influxdata/ui/blob/master/src/cloud/utils/reporting.ts#L78

and here is the lodash version many libraries implement which calls a function at MOST every X ms which is good for things like tracking fast moving pointer events without clobbering a render cycle:
https://github.com/lodash/lodash/blob/master/debounce.js

you can even rewrite one of those to connect to the next frame render with `window.requestAnimationFrame` instead of chosing an arbitrary time to decide your debounce frequency! you'll find that version more in rendering libraries and the like

Reply:
Good to know! Even though I'm still digesting on this :)